generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -84,6 +85,134 @@ public OpaqueExpressionGenerator(IRandom generator, GenerationParams generationP
 
   }
 
+  @FunctionalInterface
+  public interface OpaqueZeroOneFactory {
+    Optional tryMakeOpaque(BasicType type, boolean constContext, final int depth,
+                           Fuzzer fuzzer, boolean isZero);
+
```

Comment:
I suggest having a method "waysToMakeZeroOrOne" that returns the list of common ways, and having each of "waysToMakeZero" and "waysToMakeOne" return a list comprised of the common ways, plus any additional more specific ways.  This will avoid the problem of forgetting to add a new common way in both places.